ember-core/src/test/scala/org/http4s/ember/core/StreamingParserSuite.scala

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2019 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.ember.core
+
+import cats.effect._
+import cats.syntax.all._
+import fs2._
+import fs2.concurrent.Queue
+import org.http4s.Http4sSuite
+import org.scalacheck.Gen
+import org.scalacheck.Prop._
+import org.scalacheck.effect.PropF
+
+class StreamingParserSuite extends Http4sSuite {
+
+  object Helpers {
+    def taking[F[_]: Concurrent, A](segments: List[List[A]]): F[F[Option[Chunk[A]]]] =
+      for {
+        q <- Queue.unbounded[F, Option[Chunk[A]]]
+        _ <- segments.traverse(bytes => q.enqueue1(Some(Chunk.seq(bytes))))
+        _ <- q.enqueue1(None)
+      } yield q.dequeue1
+
+    def subdivided[A](as: List[A], count: Int): Gen[List[List[A]]] = {
```

Comment:
Given a list, subdivide it up to `count` times such that the concatenation of all the sublists of the returned list equals the original list (there is a property-based test that asserts this at the end of the file). Maybe we can upstream this back into scalacheck?

Reply:
It may be a little esoteric for scalacheck, but it's cool.  It seems useful in fs2 applications, at least.  They don't publish a testkit, I don't think?

Reply:
AFAIK they don't